### PR TITLE
Local dev server uses bundled key/cert pair

### PR DESCRIPTION
**Pull request by johnboxall**
Chrome 58 dropped support for CN validation in HTTPS:

https://textslashplain.com/2017/03/10/chrome-deprecates-subject-cn-matching/

Which means our previous [trick](https://github.com/mobify/platform-scaffold/tree/develop/web#prevent-ssl-errors-in-preview) no longer works.

This change adds a bundled certificate that works on Chrome 58+ and Safari.

It also includes a script to regenerate the cert if needed.

I don't think this is a perfect solution. We need a long term solution to local development over HTTPS – but this is better than what we have right now.
(brief description / PR title)

## Changes
- Add a new key/pair and have the webpack server use it
- Add a script to regenerate it

## How to test-drive this PR 🚗 
```
git checkout -t origin/add-localhost-cert
npm run dev
```
In Chrome 58+ open https://localhost:8443/:

image

Back in terminal:

```
open dev-server/localhost.pem
```

Add it to your keychain and change the settings to always trust.

In Chrome 58+ refresh https://localhost:8443/:

image

YES! 🔒

### CODE REVIEW

**Comment by ellenmobify** [comment]
👍 Works on my Chrome 58

File: web/README.md [hunk 1]
```diff
@@ -20,49 +20,43 @@ npm run save-credentials -- -u <myEmail@organization.com> -k <myAPIkey>
...
-To add the certificate to the Mac system trust store and make the
-browsers accept it, do the following:
+To add the certificate to the Mac system trust store:
 
-1. In the root of the project directory, run `open node_modules/webpack-dev-server/ssl/server.crt`.
-2. Open `Keychain Access` -> go to `Certificates` -> select `localhost`.
-3. Right click on the entry and select `Get Info`.
+0. Open https://localhost:8443. *You should see a security warning.* ⚠️
+1. In the root of the project directory, run `open dev-server/localhost.pem`.
+2. Add the certifcate to your `login` Keychain.
```
Comment (jeremywiebe) [inline web/README.md]: "certificate" (spelling)

File: web/README.md [hunk 2]
```diff
@@ -20,49 +20,43 @@ npm run save-credentials -- -u <myEmail@organization.com> -k <myAPIkey>
...
+2. In `Keychain Access` open `Certificates` and find `Mobify Development Server`.
```
Comment (jeremywiebe) [inline web/README.md]: List has two "2"s in it.

File: web/README.md [hunk 3]
```diff
@@ -20,49 +20,43 @@ npm run save-credentials -- -u <myEmail@organization.com> -k <myAPIkey>
...
+0.  Open https://localhost:8443. *You should see a security warning.* ⚠️
+1.  Start Menu → Run `mmc.exe`.
+2.  File → Add/Remove Snap-in.
+3.  Select "Certificates" and click Add.
+4.  Select "Computer Account" and click Next.
+5.  Select "Local Computer" and click Finish.
+6.  Click OK to close the Add or Remove Snap Ins dialog.
+7.  Expand the Certificates node and right-click on the Trusted Roots Certification Authorities node.
+8.  Select All Tasks → Import.
+9.  Import the file at `$\web\dev-server\localhost.pem`. Leave all other settings as is while importing.
```
Comment (jeremywiebe) [inline web/README.md]: Windows paths have `/` (forward slashes)

File: web/scripts/create-localhost-certificate.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+x509_extensions    = x509_ext
+
+[ subject ]
+commonName = $COMMON_NAME
+
+[ x509_ext ]
+subjectAltName = @alternate_names
+
+[ alternate_names ]
+DNS.1 = localhost
```
Comment (jeremywiebe) [inline web/scripts/create-localhost-certificate.sh]: 🎉

File: web/README.md (hunk 3 above)
Comment (jeremywiebe) [inline web/README.md]: Oops.. it's too early... you're right... unix uses `/`  :(

**Comment by johnboxall** [comment]
Thanks for the feedback @ellenmobify & @jeremywiebe.

**Pull request merged**